kratos/tests/cpp_tests/geometries/test_brep_surface.cpp

```diff
@@ -195,5 +195,63 @@ namespace Testing {
         //auto results = p_brep_curve_on_surface.GlobalDerivatives(coords, 3);
         //KRATOS_WATCH(results[0])
     }
+
+    KRATOS_TEST_CASE_IN_SUITE(NurbsBrepSurfaceSurrogate, KratosCoreGeometriesFastSuite)
+    {
+        auto p_surface = GenerateReferenceNodeSurfaceHalfCirclePointer();
+
+        using BrepSurfaceType = BrepSurface<PointerVector<Node>, true, PointerVector<Point>>;
+        using BrepCurveOnSurfaceType = BrepCurveOnSurface<PointerVector<Node>, true, PointerVector<Point>>;
```

Comment:
```suggestion
        using BrepCurveOnSurfaceType = typename BrepSurfaceType::BrepCurveOnSurfaceType;
```

Reply:
If we eventually change the type in `BrepSurfaceType` this will automatically handle it.